src/dotnet/commands/dotnet-restore3/Program.cs

```diff
@@ -60,6 +60,12 @@ public static int Run(string[] args)
                     "Treat package source failures as warnings.",
                     CommandOptionType.NoValue);
 
+            // Use a boolean argument instead of a switch to match nuget.exe
```

Comment:
Not currently but I could add some integration tests. The amount of logic is minimal in this wrapper, everything else is tested on the nuget side.